### PR TITLE
Mongodb compass

### PR DESCRIPTION
**Pull request by mijorus**
- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ ] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [ ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by mijorus** [comment]
Please ignore the branch history, I checked out from an existing branch

**Comment by mijorus** [comment]
I'm also implementing `x-checker-data` in the next commits

File: com.mongodb.Compass.json [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    "base-version": "21.08",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "21.08",
+    "sdk": "org.freedesktop.Sdk",
+    "sdk-extensions": [
+        "org.freedesktop.Sdk.Extension.node14"
+    ],
+    "command": "run.sh",
+    "separate-locales": false,
+    "tags": [],
```
Comment (hfiguiere) [inline com.mongodb.Compass.json:13]:
```suggestion
```

File: com.mongodb.Compass.json [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+                "install -Dm 644 icon-256x256.png /app/share/icons/hicolor/256x256/apps/com.mongodb.Compass.png",
+                "install -Dm 644 mongodb-compass.desktop /app/share/applications/com.mongodb.Compass"
+            ],
+            "sources": [
+                {
+                    "type": "archive",
+                    "only-arches": [
+                        "x86_64"
+                    ],
+                    "url": "https://github.com/mongodb-js/compass/releases/download/v1.32.2/mongodb-compass-1.32.2-linux-x64.tar.gz",
```
Comment (hfiguiere) [inline com.mongodb.Compass.json:61]: since this is x86_64 only you need to instruct flathub to not build it

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: com.mongodb.Compass.json [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        },
+        {
+            "name": "compass",
+            "buildsystem": "simple",
+            "build-commands": [
+                "cp -r main /app/main",
+                "install -Dm 644 icon-64x64.png /app/share/icons/hicolor/64x64/apps/com.mongodb.Compass.png",
+                "install -Dm 644 icon-128x128.png /app/share/icons/hicolor/128x128/apps/com.mongodb.Compass.png",
+                "install -Dm 644 icon-256x256.png /app/share/icons/hicolor/256x256/apps/com.mongodb.Compass.png",
+                "install -Dm 644 mongodb-compass.desktop /app/share/applications/com.mongodb.Compass"
```
[the remark this answers is not in the document]
Comment (mijorus) [reply com.mongodb.Compass.json:58]: adding everything RN

File: com.mongodb.Compass.json [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    "sdk": "org.freedesktop.Sdk",
+    "sdk-extensions": [
+        "org.freedesktop.Sdk.Extension.node14"
+    ],
+    "tags": [
+    	"mongodb",
+    	"database",
+    	"db",
+    	"nosql"	
+    ],
```
Comment (hfiguiere) [inline com.mongodb.Compass.json:16]:
```suggestion
```

File: mongodb-compass.desktop [hunk 5]
```diff
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Name=MongoDB Compass
+Comment=The MongoDB GUI
+GenericName=MongoDB Compass
+Exec=mongodb-compass %U
+Icon=mongodb-compass
```
Comment (hfiguiere) [inline mongodb-compass.desktop:6]: this should be the app-id

File: com.mongodb.Compass.json [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+        },
+        {
+            "name": "compass",
+            "buildsystem": "simple",
+            "build-commands": [
+                "cp -r main /app/main",
+                "install -Dm 644 icon-64x64.png /app/share/icons/hicolor/64x64/apps/com.mongodb.Compass.png",
+                "install -Dm 644 icon-128x128.png /app/share/icons/hicolor/128x128/apps/com.mongodb.Compass.png",
+                "install -Dm 644 icon-256x256.png /app/share/icons/hicolor/256x256/apps/com.mongodb.Compass.png",
+                "install -Dm 644 mongodb-compass.desktop /app/share/applications/com.mongodb.Compass"
```
Comment (hfiguiere) [inline com.mongodb.Compass.json:58]:
```suggestion
                "install -Dm 644 mongodb-compass.desktop /app/share/applications/com.mongodb.Compass.desktop"
```

File: com.mongodb.Compass.json (hunk 2 above)
Comment (mijorus) [reply com.mongodb.Compass.json:66]: aarch64 build is available as `.deb` package

https://github.com/mongodb-js/compass/releases/tag/v1.32.2

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: com.mongodb.Compass.appdata.xml [hunk 7]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+    <id>com.mongodb.Compass</id>
+    <metadata_license>MIT</metadata_license>
+    <project_license>LicenseRef-proprietary=SSPL-1.0</project_license>
```
Comment (hfiguiere) [inline com.mongodb.Compass.appdata.xml:5]:
```suggestion
    <project_license>SSPL-1.0</project_license>
```

File: com.mongodb.Compass.json (hunk 2 above)
Comment (hfiguiere) [reply com.mongodb.Compass.json:60]: I wish you were right here, but I don't see it. As for unpacking .deb, there are a lot of flatpak that do that.

You could build from source though.

File: com.mongodb.Compass.json [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+                    "only-arches": [
+                        "x86_64"
+                    ],
+                    "url": "https://github.com/mongodb-js/compass/releases/download/v1.32.2/mongodb-compass-1.32.2-linux-x64.tar.gz",
+                    "sha256": "d69d97c9a0ea26c5602147b6467fd6aeb4d7e20e8feefd4a03d2639d58b89dfe",
+                    "dest": "main"
+                },
+                {
+                    "type": "file",
+                    "path": "./icons/icon-256x256.png"
```
[the remark this answers is not in the document]
Comment (mijorus) [reply com.mongodb.Compass.json:66]: icons have been resized by me using `convert`

[the remark this answers is not in the document]
Comment (hfiguiere) [reply com.mongodb.Compass.json:66]: no SVG?

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: com.mongodb.Compass.appdata.xml (hunk 7 above)
Comment (mijorus) [reply com.mongodb.Compass.appdata.xml:5]: `appstream-util` (on my pc) complains

Comment (mijorus) [reply com.mongodb.Compass.appdata.xml:5]:
```
• tag-invalid           : <project_license> is not valid [SSPL-1.0]SPDX ID 'SSPL-1.0' unknown
```

**Comment by mijorus** [comment]
mongodb ships with a png by default

File: com.mongodb.Compass.json (hunk 2 above)
Comment (mijorus) [reply com.mongodb.Compass.json:60]: I'll give a look, thanks

File: mongodb-compass.desktop [hunk 9]
```diff
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Name=MongoDB Compass
+Comment=The MongoDB GUI
+GenericName=MongoDB Compass
+Exec=com.mongodb.Compass %U
```
Comment (hfiguiere) [inline mongodb-compass.desktop:5]: this should match the `command`.

File: com.mongodb.Compass.json [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+                    ]
+                }
+            ]
+        },
+        {
+            "name": "compass",
+            "buildsystem": "simple",
+            "build-commands": [
+                "cp -r main /app/main",
+                "install -Dm 644 icon-64x64.png /app/share/icons/hicolor/64x64/apps/com.mongodb.Compass.png",
```
Comment (hfiguiere) [inline com.mongodb.Compass.json:49]: note that we only require 128x128 (or SVG). And given they are not provided upstream, I'd be enclined to says 128x128 is the only one we need to have.

Comment (mijorus) [reply com.mongodb.Compass.json:49]: not what this says:
https://github.com/flathub/flathub/wiki/App-Requirements#application-icons

btw I have included the two required sized plus the original icon untouched

Comment (hfiguiere) [reply com.mongodb.Compass.json:49]: yeah **64x64 only** is no longer accepted. But I can't edit the wiki, so it's just a lie now.

**Comment by hfiguiere** [comment]
bot, build com.mongodb.Compass

**Comment by mijorus** [comment]
bot, build com.mongodb.Compass

**Comment by mijorus** [comment]
bot, build com.mongodb.Compass

**Comment by mijorus** [comment]
Please wait, I found a bug

**Comment by mijorus** [comment]
Help is welcome here, it seems that Compass won't save DB passwords where it is supposed to, or maybe these passwords are not stored correctly

**Comment by hfiguiere** [comment]
where is it supposed to save them?

**Comment by hfiguiere** [comment]
there is not filesystem permissions, so if it doesn't use XDG standard directories, then it's likely nothing is persisted.

**Comment by barthalion** [comment]
Looks fine here, as long as I save the connection as a favourite.

> I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub.

Please contact upstream about your submission.

**Comment by barthalion** [comment]
The password seems to disappear after I close the app, yeah.

**Comment by mijorus** [comment]
Launching with:

```
flatpak run -vv --env=NODE_ENV=development --env=DEBUG='*' com.mongodb.Compass 
```

enables unlocks many more logs but nothing interesting shows up😔

**Comment by mijorus** [comment]
bot, build com.mongodb.Compass

**Comment by mijorus** [comment]
https://github.com/flathub/flathub/pull/3269/commits/3427ca6a13956b8b1bd4ca8fd5b321f40b8aeada

this worked, source: https://github.com/flathub/com.microsoft.Teams/pull/48/commits/74040c479beaf325f7b7c95210c74ba90d7d3026

**Pull request closed without merging**